packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js

```diff
@@ -30,4 +47,42 @@ export default class SoftphoneController {
     }
     this.osName = MOBILE_OS.windows;
   }
+
+  updateFilesList() {
+    if (this.fileModel?.length > 1) {
+      this.fileModel.shift();
+    }
+  }
+
+  applyTheme() {
+    this.softphoneService
+      .uploadDocument(this.fileModel[0])
+      .then((url) => {
+        this.softphoneService
+          .putSoftphoneTheme(
+            this.$stateParams.billingAccount,
```

Comment:
You can use resolve on routing to pass thoses params and not inject stateParams into the controller

Reply:
Indeed